doc/developer-notes.md

```diff
@@ -439,6 +439,14 @@ General C++
 
   - *Rationale*: This avoids memory and resource leaks, and ensures exception safety
 
+- Do not rely on unsigned wrap-around semantics unless there are good reasons for doing so (e.g. hash functions).
+  Functions that intentionally rely on unsigned wrap-around semantics should be annotated using `ALLOW_WRAPAROUND`.
+  Functions that unintentionally trigger unsigned wrap-arounds should be annotated using
+  `WARNING_UNINTENTIONAL_WRAPAROUND` while awaiting being fixed.
```

Comment:
would it be unrealistic to fix all these before landing this patch, so this annotation is not needed?

Reply:
@arvidn I'm not sure TBH: the review activity in #1153 which fixes these cases has been quite low during the year. My main goal with this PR is to make it less likely that we introduce new unintentional wraparounds going forward :-)